### PR TITLE
Feat/actions

### PR DESCRIPTION
**Pull request by Ankit-clouddrove**
## what
* Added RemoteSSHCommadn.yml file for running the bash commands on Ec2 using the private SSH key and Send Slack Notifications after the execution of Action workflow file.
* Added Readme file for RemoteSSHCommand.yml file. 
* Added AWSSSMSendCommand.yml file for running the bash commands on Ec2 without SSH and Send slack Notifications after the execution of Action workflow file.
* Added Readme file for AWSSSMSendCommand.yml file.
* Added Dockerfile path input in docker-scanner.yml file.
* Added Security upload input in the docker-scanner.yml file with condition statement.

## why
* To create a YAML file RemoteSSHCommand.yml that defines a workflow for running Bash commands on an EC2 instance using a private SSH key.This is a part of automation process where commands need to be executed on a remote server securely via SSH.
* Create Readme file RemoteSSHCommand.md provides documentation and instructions for using the RemoteSSHCommand.yml workflow file.
* Create a YAML file AWSSSMSendCommand.yml that defines a workflow for running Bash commands on an EC2 instance without using SSH. Instead, AWS Systems Manager (SSM) is used for command execution.AWS SSM allows for remote command execution without requiring direct SSH access. This could be useful in scenarios where SSH access is restricted or not desired.
* Create Readme file AWSSSMSendCommand.md provides documentation and instructions for using the AWSSSMSendCommand.yml workflow file.
* To introduce a new input parameter "dockerfile-path" in the docker-scanner.yml workflow file, allowing users to specify the path to the Dockerfile.This input parameter provides flexibility to users who want to customize the location of their Dockerfile within the project structure.
* To include an input parameter "security-upload" in the docker-scanner.yml workflow file, is used  to uploading security-related information to GitHub security only if certain checks or conditions are met.

### CODE REVIEW

File: .github/workflows/RemoteSSHCommand.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        default: "30s"
+        required: false
+      command_timeout:
+        description: "Timeout duration for executing SSH commands."
+        type: string
+        default: "10m"
+        required: false
+      script:
+        description: "Specify the commands to be executed on the remote host."
+        type: string
```
Comment (VishwajitNagulkar) [inline .github/workflows/RemoteSSHCommand.yml:26]: Add `required` attribute here

File: .github/workflows/RemoteSSHCommand.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        default: false
+      SLACK_USERNAME:
+        description: "It is the name displayed to others in Message on Slack channel"
+        required: true
+        type: string
+      SLACK_FOOTER:
+        description: "Additional information or context often placed at the bottom of a message in Slack"
+        required: true
+        type: string
```
Comment (VishwajitNagulkar) [inline .github/workflows/RemoteSSHCommand.yml:53]: change required value to the false as slack notification step is optional

File: .github/workflows/AWSSSMSendCommand.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          working-directory: ${{ inputs.working-directory }}
+          command: |-
+            ${{ inputs.command }}
+
+      - name: Slack notification
+        if: ${{ inputs.slack-notification == 'true' && always() }}
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_URL }}
+          SLACK_MESSAGE: ${{ job.status }}
```
Comment (VishwajitNagulkar) [inline .github/workflows/AWSSSMSendCommand.yml:78]: you passed `SLACK_MESSAGE` attribute as a inputs in this workflow but here you put its value hardcoded

File: .github/workflows/RemoteSSHCommand.yml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+          key: ${{ secrets.PRIVATE_SSH_KEY }}
+          port: ${{ inputs.port }}
+          envs: ${{ inputs.envs }}
+
+      - name: slack notification
+        if: ${{ inputs.slack-notification == 'true' && always() }}
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_URL }}
+          SLACK_MESSAGE: ${{ job.status }}
```
Comment (VishwajitNagulkar) [inline .github/workflows/RemoteSSHCommand.yml:105]: Same here for `SLACK_MESSAGE`

File: .github/workflows/RemoteSSHCommand.yml (hunk 1 above)
Comment (Ankit-clouddrove) [reply .github/workflows/RemoteSSHCommand.yml:27]: Attribute added

File: .github/workflows/RemoteSSHCommand.yml (hunk 2 above)
Comment (Ankit-clouddrove) [reply .github/workflows/RemoteSSHCommand.yml]: required value change to false done

File: .github/workflows/RemoteSSHCommand.yml (hunk 4 above)
Comment (Ankit-clouddrove) [reply .github/workflows/RemoteSSHCommand.yml]: Added default value in inputs and set the SLACK_MESSAGE as input

File: .github/workflows/AWSSSMSendCommand.yml (hunk 3 above)
Comment (Ankit-clouddrove) [reply .github/workflows/AWSSSMSendCommand.yml]: Added default value as job.status  in inputs and set the SLACK_MESSAGE as inputs.slack_message

**Review by VishwajitNagulkar — APPROVED** [review]
LGTM!

**Review by VishwajitNagulkar — APPROVED** [review]

**Review by d4kverma — APPROVED** [review]

**Review by themaniskshah — APPROVED** [review]
LGTM!

**Pull request merged**